packages/open-next/src/core/routing/middleware.ts

```diff
@@ -54,8 +54,10 @@ export async function handleMiddleware(
   // We bypass the middleware if the request is internal
   if (internalEvent.headers["x-isr"]) return internalEvent;
 
+  const protocol = new URL(internalEvent.url).protocol;
```

Comment:
We can't do that here, in lambda `url` only contains the rawPath and the query, no protocol or host
We could use https://github.com/opennextjs/opennextjs-aws/blob/811bdc0008213eed9f859fc505c92d85327a1bac/packages/open-next/src/core/routing/util.ts#L44 along with isExternal https://github.com/opennextjs/opennextjs-aws/blob/811bdc0008213eed9f859fc505c92d85327a1bac/packages/open-next/src/core/routing/util.ts#L19

If not found we default to `https`, would this work for you ?